### DISCUSSION TITLE
Usable as multi-select with checkboxes?

### DISCUSSION THREAD

**Discussion by fre-ben** [lukasbach/react-complex-tree#16] [Q&A]
Hey there!
Has anyone ever used this component as multi-select with checkboxes?
So that when you tick the checkbox of a parent all the child elements/checkboxes get selected as well?

**Comment by lukasbach** [comment]
Not tried that out, but you can try to listen to select events (`onSelectItems`) and then manually adjusting the selected state by adding the childs as selected items (see https://rct.lukasbach.com/docs/guides/refs)

**Comment by ronnyek** [comment]
I too want to do this... seems like one could override renderer, and maybe use a custom interaction mode? I think the key for me, is that clicking labels, or expanding nodes does not change tree selection. 

I've been able to simply copy default renderer for items, and add a checkbox with the `checked = {context.isChecked}` which works, but doesn't let you deselect (because I'm not handling the deselection in clicking of the checkbox (I also think we also need to disable the default selection behavior)

If this is something that should be possible, I'd like to implement it the most correct way.

(Btw, I actually don't have a problem with the default functionality, I realize ctrl and shift modifiers while clicking do allow multi-selection, I just am building an app where users aren't going to know that)
